fix(DebugRecorder): Create parent folders & Fallback log writing to clipboard

sqlerrorthing opened the pull request:
![image](https://github.com/user-attachments/assets/047c55f5-f049-4ad3-bcc2-2424d28139e7)

сука, я на это ЧАС потратил, потому что какой то долбаеб не добавил возможности записи лога в буфер обмена, Я ВАЩЕ не ебу почему не получилось записать в файл, мб папки небыло, поэтому добавил и создание этой же папки.
нет ну это вообще пиздец, изуна.

я час проебал на то что бы сделать ахуенную модель, но мне КАКОГО ТО ХУЯ поводили хуем полбу

sqlerrorthing committed:
fix/debug-recorder-log-writing

sqlerrorthing commented:
https://github.com/user-attachments/assets/796a7642-8fc9-4b6e-9164-413f98770422

MukjepScarlet commented:
This is not needed. The critical problem is that the `toJson` function returns a huuuge String. My solution is to use `toJson` directly on a `Writer` of the file.

sqlerrorthing commented:
> This is not needed. The critical problem is that the `toJson` function returns a huuuge String. My solution is to use `toJson` directly on a `Writer` of the file.

no, error in my case - i deleted the folder, fixed by `runCatching { file.parentFile.mkdirs() }`

sqlerrorthing commented:
> The critical problem is that the toJson function returns a huuuge String.

и? оно умеет работать с большими строками. нет, не в этом проблема ТОЧНО.
или проведи тесты самостоятельно, или просто погугли ограничения.
майнкрафт залагивает когда идет сохранение, да, там очень очень большая строка.
но jvm способна ее обработать, как и gson создать ее.

sqlerrorthing committed:
buffered writer

sqlerrorthing committed:
simple buffered writer

MukjepScarlet commented:
> > This is not needed. The critical problem is that the `toJson` function returns a huuuge String. My solution is to use `toJson` directly on a `Writer` of the file.
> 
> no, error in my case - i deleted the folder, fixed by `runCatching { file.parentFile.mkdirs() }`

Ah I noticed your fix

MukjepScarlet commented on src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/debugrecorder/ModuleDebugRecorder.kt:
```diff
@@ -64,10 +68,24 @@ object ModuleDebugRecorder : ClientModule("DebugRecorder", Category.MISC) {
                 chat(markAsError("Failed to write log to file $it".asText()))
+
+                runCatching {
+                    val selection = StringSelection(fileGson.toJson(this.packets))
```
The system clipboard probably can't have a 50MiB text

sqlerrorthing replied:
размер данных ограничен только объемом доступной оперативной памяти.
за исключением браузеров, там максимум до 4-рех мегабайт можно положить в буфер обмена.

1zun4 commented on src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/debugrecorder/ModuleDebugRecorder.kt:
```diff
@@ -64,10 +68,24 @@ object ModuleDebugRecorder : ClientModule("DebugRecorder", Category.MISC) {
             }.onFailure {
                 chat(markAsError("Failed to write log to file $it".asText()))
+
+                runCatching {
+                    val selection = StringSelection(fileGson.toJson(this.packets))
+                    Toolkit.getDefaultToolkit().systemClipboard.setContents(selection, null)
+                }.onFailure {
+                    chat(markAsError("Failed to write log to clipboard".asText()))
+                }.onSuccess {
+                    chat(regular("Log was written to clipboard."))
+                }
```
This is bad

sqlerrorthing replied:
why?

1zun4 replied:
Copying to the clipboard is pointless, and we are generating another JSON for something we are not going to use.
